### PR TITLE
Require code/run-time dependencies as host dependencies

### PR DESCRIPTION
**Pull request by Luthaf**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] ~Reset the build number to `0` (if the version changed)~
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

As mentioned in https://github.com/conda-forge/admin-requests/pull/115#issuecomment-678697039, we need to request these dependencies as host deps to allow cross-compiling.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -60,6 +60,7 @@ requirements:
     - ninja
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
+  host:
     - libnetcdf >=4.7.4
```
Comment (isuruf) [inline recipe/meta.yaml]:
```suggestion
    - libnetcdf
```

Comment (Luthaf) [reply recipe/meta.yaml]: Will the resulting .so on linux require `libnetcdf.so` instead of the specific `libnetcdf.so.18`?

Comment (Luthaf) [reply recipe/meta.yaml]: Or should I remove the requirement only in the `host` section and not the `run` section?

Comment (isuruf) [reply recipe/meta.yaml]: It'll anyway require `libnetcdf.so.18`. Libraries like `libnetcdf` are pinned globally in conda-forge and when we upgrade the version, you'll get an automated PR to rebuild chemfiles-lib.

**Pull request merged**